exporters/prometheus/exporter.go

```diff
@@ -120,8 +133,18 @@ type metricData struct {
 	histogramBuckets map[float64]uint64
 }
 
-func getMetricData(metrics metricdata.ResourceMetrics) []*metricData {
+func (c *collector) getMetricData(metrics metricdata.ResourceMetrics) []*metricData {
 	allMetrics := make([]*metricData, 0)
+
+	if !c.disableTargetInfo {
+		if c.targetInfo == nil {
+			// Resource should be immutable, we don't need to compute again
+			c.targetInfo = createInfoMetricData(targetInfoMetricName, targetInfoDescription, metrics.Resource)
```

Comment:
Should a sync.Once be used here instead to ensure this is concurrent safe?

Reply:
Yeah, that is way better! Changed to use a `sync.Once` here :)